hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -191,8 +188,6 @@ private void validateSemantics(
                 validateFalse(
                         entitiesConfig.limitTokenAssociations() && newMaxAssociations > tokensConfig.maxPerAccount(),
                         REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT);
```

Comment:
Is there any possibility that `newMaxAssociations` is negative in this block because `entitiesConfig.unlimitedAutoAssociationsEnabled()` is false?  Or do the other validations in this block preclude this from happening?

Reply:
We actually check for negative here: https://github.com/hashgraph/hedera-services/blob/ceb6020b1511031ab6fa9cdab9918396c05c0619/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java#L179

And test it with HapiTest
https://github.com/hashgraph/hedera-services/blob/ceb6020b1511031ab6fa9cdab9918396c05c0619/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractUpdateSuite.java#L503

Reply:
Right I saw that I was wondering what would happen if the first clause was false and the `newMaxAssociations < 0` is true?  Not a big deal this is really just a nit.

Reply:
I see what you meant. Thanks! I added additional check for negative.